apps/server/src/modules/provisioning/strategy/iserv/iserv.strategy.ts

```diff
@@ -1,25 +1,34 @@
-import { ProvisioningStrategy } from '@src/modules/provisioning/strategy/base.strategy';
 import { Injectable } from '@nestjs/common';
 import { SystemProvisioningStrategy } from '@shared/domain/interface/system-provisioning.strategy';
-import jwt, { JwtPayload } from 'jsonwebtoken';
-import { ProvisioningDto } from '@src/modules/provisioning/dto/provisioning.dto';
 import { OAuthSSOError } from '@src/modules/oauth/error/oauth-sso.error';
-
-export type IservStrategyData = {
-	idToken: string;
-};
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { ExternalUserDto, OauthDataDto, OauthDataStrategyInputDto, ProvisioningDto } from '../../dto';
+import { ProvisioningStrategy } from '../base.strategy';
 
 @Injectable()
-export class IservProvisioningStrategy extends ProvisioningStrategy<IservStrategyData> {
-	override async apply(params: IservStrategyData): Promise<ProvisioningDto> {
-		const idToken: JwtPayload | null = jwt.decode(params.idToken, { json: true });
+export class IservProvisioningStrategy extends ProvisioningStrategy {
+	getType(): SystemProvisioningStrategy {
+		return SystemProvisioningStrategy.ISERV;
+	}
+
+	override getData(input: OauthDataStrategyInputDto): Promise<OauthDataDto> {
+		const idToken: JwtPayload | null = jwt.decode(input.idToken, { json: true });
 		if (!idToken || !idToken.uuid) {
-			throw new OAuthSSOError('Failed to extract uuid', 'sso_jwt_problem');
+			return Promise.reject(new OAuthSSOError('Failed to extract uuid', 'sso_jwt_problem'));

```

Comment:
Why?

Reply:
This is a synchronous function that has to return a promise (in this case). When using throw in a synchronous function no promise will be created. In order to create a rejected promise I use `Promise.reject`. An alternative would be to wrap the whole function in a promise like:
```
override getData(input: OauthDataStrategyInputDto): Promise<OauthDataDto> {
  return new Promise((resolve, reject) => {
  ...
  reject(new OAuthSSOError(...)); or throw new OAuthSSOError(...);
  ...
  });
}
```